server/models/Transaction.js

```diff
@@ -476,6 +476,13 @@ export default (Sequelize, DataTypes) => {
       return;
     }
 
+    // Calculate the paymentProcessorFee proportional to the feeOnTop amount
+    const feeOnTopPaymentProcessorFee = toNegative(
+      Math.round(
+        transaction.paymentProcessorFeeInHostCurrency *
+          Math.abs(transaction.platformFeeInHostCurrency / transaction.amountInHostCurrency),
+      ),
+    );
```

Comment:
When doing math in code, I personally find that using as many variables as possibles improves clarity by naming things.  But that's a personal preference

```suggestion
    const feeOnTopPercent = Math.abs(transaction.platformFeeInHostCurrency / transaction.amountInHostCurrency);
    const feeOnTopPaymentProcessorFee = toNegative(
      Math.round(transaction.paymentProcessorFeeInHostCurrency * feeOnTopPercent),
    );
```